ConsoleBank/Currencies/CurrenciesApi.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Currencies.Entities;
+using Flurl;
+using Flurl.Http;
+
+namespace Currencies
+{
+    internal class CurrenciesApi : ICurrenciesApi
+    {
+        private const string CurrencyRatesApiUrl = "https://www.nbrb.by/api/exrates/rates";
+        private const string CurrenciesApiUrl = "https://www.nbrb.by/api/exrates/currencies";
+        private readonly List<CurrencyRate> _currentyRates = new();
+        private int _now;
+
+        public Task<Currency[]> GetCurrencies()
+        {
+            return CallApi(() => CurrenciesApiUrl.GetJsonAsync<Currency[]>());
+        }
+
+        public async Task<CurrencyRate> GetCurrencyRate(int currencyId, DateTime? ondate)
+        {
+            if (!ondate.HasValue)
+            {
+                if (_now != DateTime.Now.DayOfYear)
+                {
+                    await SetListCurrencyRates();
+                }
+
+                return await Task.FromResult(_currentyRates.Where(x => x.Id == currencyId).FirstOrDefault());
+            }
+
+            return await CallApi(() => CurrencyRatesApiUrl
+                    .AppendPathSegment(currencyId)
+                    .SetQueryParams(new { ondate = ondate?.ToString("yyyy-MM-dd") })
```

Comment:
мы проверили на null выше
```suggestion
                    .SetQueryParams(new { ondate = ondate.ToString("yyyy-MM-dd") })
```

Reply:
в силе